### PR TITLE
[Shallow] Aux scripts for coupling posprocess

### PR DESCRIPTION
**Pull request by miguelmaso**
**📝 Description**
Fix a bad copy between model parts and add a new file with auxiliary procedures.

**🆕 Changelog**
- Fix copy variable
- Add new file with auxiliary procedures

### CODE REVIEW

**Review by ipouplana — DISMISSED** [review]

File: applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    return post_files
+
+
+def _IsGiDFile(file_name):
+    if file_name.endswith(_binary_extension):
+        return True
+    elif file_name.endswith(_ascii_extension):
+        return True
+    else:
+        return False
```
Comment (philbucher) [inline applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py]:
```suggestion
    return file_name.endswith(_binary_extension) or file_name.endswith(_ascii_extension)
```

File: applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+def _CheckFileName(file_name):
+    if not file_name:
+        file_name = path.basename(getcwd())
+    if not file_name.endswith(_list_extension):
+        file_name += _list_extension
+    return file_name
+
+
+def _WriteListFile(file_list, file_name):
+    file = open(file_name, "w")
```
Comment (philbucher) [inline applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py]: I recommend using `with open(...)`

File: applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+from os import path, listdir, getcwd
```
Comment (philbucher) [inline applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py]: I recommend checking out the [pathlib](https://docs.python.org/3/library/pathlib.html) lib which makes those things easier / more native

File: applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+_list_extension = ".post.lst"
+_ascii_extension = ".post.res"
+_binary_extension = ".post.bin"
+
+def GenerateGiDListFile(first_dir = None, other_dirs = [], file_name = None):
```
Comment (philbucher) [inline applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py]: note that it is dangerous to use objects as default args
E.g. using this in CTors of Classes will result in the same object being used!
It is one of the nastier pitfalls of python

**Review by ipouplana — DISMISSED** [review]

File: applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py (hunk 4 above)
Comment (miguelmaso) [reply applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py]: thanks for the hint

**Review by ipouplana — DISMISSED** [review]

File: applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py (hunk 3 above)
Comment (miguelmaso) [reply applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py]: pathlib is really nice! Thanks

File: applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py (hunk 2 above)
Comment (miguelmaso) [reply applications/ShallowWaterApplication/python_scripts/postprocess/generate_gid_list_file.py]: done

**Comment by miguelmaso** [comment]
@ipouplana now its ready to merge. In the future this file will be a regular process

**Review by ipouplana — APPROVED** [review]

**Review by philbucher — APPROVED** [review]
oh wow nice!

(btw my comment was more informative, didn't mean that you should have changed it right away. I am only trying to inform ppl if I see sth that can be done in a different/better/newer way)

**Pull request merged**